versions/1.21.5/src/main/java/at/hannibal2/skyhanni/features/foraging/ForagingTracker.kt

```diff
@@ -236,22 +259,59 @@ object ForagingTracker : SkyHanniBucketedItemTracker<ForagingTrackerLegacy.TreeT
             }
         }
 
-        if (!openBonusGiftLoop) return
-        ForagingTrackerLegacy.bonusGiftRewardPattern.matchMatcher(message) {
-            val item = group("item")
-            val itemInternalName = ForagingTrackerLegacy.enchantedBookPattern.matchMatcher(item) {
-                val book = group("book")
-                val tier = group("tier").romanToDecimal()
-                NeuInternalName.fromItemNameOrNull("$book $tier")
-            } ?: NeuInternalName.fromItemNameOrNull(item) ?: return@matchMatcher
-
-            // Stretching Sticks handled separately due to stack size not being given in message
-            if (itemInternalName == STRETCHING_STICKS) return@matchMatcher
+        if (openBonusGiftLoop) {
+            ForagingTrackerLegacy.bonusGiftRewardPattern.matchMatcher(message) {
```

Comment:
This regex workaround here here should eventually (outside of this PR scope probably) get moved into `NeuInternalName` or `ItemNameResolver` to function generically